pkg/serving/config_changes.go

```diff
@@ -86,6 +86,26 @@ func UpdateImage(template *servingv1alpha1.RevisionTemplateSpec, image string) e
 	return nil
 }
 
+// UpdateContainerPort updates container with a give port
+func UpdateContainerPort(template *servingv1alpha1.RevisionTemplateSpec, port int32) error {
+	container, err := extractContainer(template)
+	if err != nil {
+		return err
+	}
+
+	if len(container.Ports) == 0 {
```

Comment:
@savitaashture : I think @rhuss mentioned to worry about the len of container ports, just replace it with `container.Ports[0].ContainerPort = port`

Reply:
@navidshaikh this condition i have added because while executing
```kn service create --image imagename --port 80```
during that time contanier.Ports is nil so iam adding that condition to check and initialize the container.Ports.

Reply:
I would just then unconditionally set

```go
container.Ports = []corev1.ContainerPort{{
			ContainerPort: port,
		}}
```

in any case. This makes the code simple to read, and we are sure that only 1 port is set (although as mentioned the slice should have only one entry)

Reply:
Updated